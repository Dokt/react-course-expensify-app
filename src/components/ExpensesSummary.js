import React from 'react';
import selectExpensesTotal from '../selectors/expenses-total.js';
import selectExpenses from '../selectors/expenses';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    const expenseCount = props.expenses.length;
    return (
        <div>
            <h3>Viewing {expenseCount} {props.expenses.length <= 1 ? 'expense' : 'expenses'} totalling {numeral(selectExpensesTotal(props.expenses)/100).format('$0,0.00')}</h3>
        </div>
    ); 
}



const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);