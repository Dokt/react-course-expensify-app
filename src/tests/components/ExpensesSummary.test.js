import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('Should render ExpensesSummary correctly with zero expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});