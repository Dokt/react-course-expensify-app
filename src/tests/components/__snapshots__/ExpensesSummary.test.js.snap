// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    3
     
    expenses
     totalling 
    $36.00
  </h3>
</div>
`;

exports[`Should render ExpensesSummary correctly with zero expenses 1`] = `
<div>
  <h3>
    Viewing 
    0
     
    expense
     totalling 
    $0.00
  </h3>
</div>
`;
